### PR TITLE
OCPBUGS-1361: Fix propagation of check errors to Available condition

### PR DESCRIPTION
**Pull request by jsafrane**
Pass overall check error to `Available` condition of Storage CR. The code was already prepared to do so, just `runChecks()` must return the actual error (or list of errors).

When at it, fix `Collect()` to actually return list of all checks instead of empty `[]checkResult`. As consequence, the detector will emit events on all check success or failures.

@openshift/storage

### CODE REVIEW

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -294,14 +294,14 @@ func (c *vSphereProblemDetectorController) runChecks(ctx context.Context, cluste
 		// (i.e. retry as slow as allowed).
 		nextDelay = defaultBackoff.Cap
 	}
-	return nextDelay, nil
+	return nextDelay, checkError
 }
```
Comment (gnufied) [inline pkg/operator/operator.go:298]: Can we have a test that checks conditions after some of the checks fail? Just a unit test is fine.

**Comment by gnufied** [comment]
The verify failure appears to be genuine:

```
pkg/operator/operator_test.go:332:11: github.com/openshift/api/operator/v1.StorageSpec composite literal uses unkeyed fields
pkg/operator/operator_test.go:337:13: github.com/openshift/api/operator/v1.StorageStatus composite literal uses unkeyed fields
```

**Comment by jsafrane** [comment]
> The verify failure appears to be genuine

Fixed

File: pkg/operator/operator.go (hunk 1 above)
Comment (jsafrane) [reply pkg/operator/operator.go:298]: Unit test added

**Pull request merged**